Yafc/Workspace/ProductionTable/ProductionLinkSummaryScreen.cs

```diff
@@ -48,11 +48,22 @@ public override void Build(ImGui gui) {
             _ = gui.BuildFactorioObjectButtonWithText(link.goods, tooltipOptions: DrawParentRecipes(link.owner, "link"));
         }
         scrollArea.Build(gui);
+        if (gui.BuildButton("Remove link", link.owner.links.Contains(link) ? SchemeColor.Primary : SchemeColor.Grey)) {
+            DestroyLink(link);
+            Close();
+        }
         if (gui.BuildButton("Done")) {
             Close();
         }
     }
 
+    private void DestroyLink(ProductionLink link) {
+        if (link.owner.links.Contains(link)) {
```

Comment:
This test isn't really necessary. Something's gone badly wrong if `Contains` returns false, and `Remove` just returns false when given something that isn't in the collection. But it does express the opinion that Contains shouldn't be returning false, so I don't think it has to be removed either.

Reply:
Are you sure? My experience with UI is JS/Browser based, and checks like this are required on double-clicks where DestroyLink can trigger twite. It's also a protection against possible later changes where i might remove the close operation.

But if my point 1 (about double clicks) cannot happen in this framework, I could leave out the check since i could always add it back when it's actually needed. Can you confirm there is zero change this code triggers twice before closing?

Reply:
I'm pretty sure DestroyLink can't be called twice (unless, as you mentioned, the Close() call is removed), but my point in that case is mostly that it's fine to call Remove twice; it only ever "fails" by returning false.